### PR TITLE
fix: edge case where empty value overwrite existing value

### PR DESCRIPTION
**Pull request by tungfinblox**
# Reason for This PR

We found another edge case where `http` package can potentially wrongly overwrite data

Let's say that we have a form like this:

```
form := url.Values{}
form.Add("name[]", "name1")
form.Add("name[]", "name2")
form.Add("name[]", "name3")
form.Add("name", "")
```

Because iterating over map in Go is random so this case might not happen all the time. But when it happens,
data under `name` key will be empty string which in our case removes all the relationships in Laravel backpack.

I also refactor the test cases to be shorter and with better description.

## Description of Changes

Fix one more edge case where empty value can potentially overwrite existing value.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the MIT license.

## PR Checklist

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.

### CODE REVIEW

**Comment by rustatian** [comment]
Hey 👋🏻, great work @tungfinblox, thanks 👍🏻
I'll have a look tomorrow ⚡

File: handler/parse.go [hunk 1]
```diff
@@ -100,46 +101,69 @@ func (dt dataTree) mount(i, v []string) error {
 	return dt[i[0]].(dataTree).mount(i[1:], v)
 }
 
-func (dt dataTree) prepareNewDataNode(i, v []string) (bool, error) {
-	_, ok := dt[i[0]]
-	if !ok {
-		dt[i[0]] = make(dataTree)
-		return true, nil
+func prepareTreeNode[T dataTree | fileTree, V []string | []*FileUpload](
```
Comment (rustatian) [inline handler/parse.go:104]: Could you please put the args into 1 line and slightly update the `hasBranch` check in the following way (to be more clear I guess):

```go
func prepareTreeNode[T dataTree | fileTree, V []string | []*FileUpload](tree T, i []string, v V) (bool, error) {
	if _, ok := tree[i[0]]; !ok {
		tree[i[0]] = make(T)
		return false, nil
	}
....
```

**Comment by rustatian** [comment]
Everything else look good (mmm generics 😆), nice job, thanks 👍🏻

**Comment by tungfinblox** [comment]
> Everything else looks good (mmm generics laughing), nice job, thanks 👍🏻

Thank you. I updated it :D
